### PR TITLE
Fix mod tools not creating new mods

### LINKED ISSUE #360: "Create a new mod" throws error when pressing OK
### Summary
If you use "Create a new mod" in the Godot Editor and press OK, an error message is printed and nothing happens.

### How to reproduce
1. Open Libre-TrainSim in the Godot Editor.
2. Open the "LTS Modding" tab.
3. Click "Create a new mod".
4. Enter the required details (unique mod name, mod display name, author).
5. Press "OK".

### Observed behaviour
The following error message is thrown (in the output tab):
```
res://addons/libre_trainsim_modding_tools/new_mod_popup.gd:9 - Invalid call. Nonexistent function 'to_valid_filename' in base 'Node (Root.gd)'.
```

### Expected behaviour
A new mod is created.

### PR DESCRIPTION
**Pull request by HaSa1002**
Also disallow the creation of mods if the unique name is empty as it would create the mod in the parent directory.

Fixes #360

### CODE REVIEW

File: src/addons/libre_trainsim_modding_tools/empty_popup.tscn [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+anchor_top = 0.5
+anchor_right = 0.5
+anchor_bottom = 0.5
+margin_left = -200.0
+margin_top = -54.5
+margin_right = 200.0
+margin_bottom = 54.5
+popup_exclusive = true
+window_title = "Information"
+dialog_text = "Please insert a valid unique mod name. Empty mod names are not allowed. Please note the automatic strippting of invalid characters."
```
Comment (nalquas) [inline src/addons/libre_trainsim_modding_tools/empty_popup.tscn]: Spotted a minor typo:
```suggestion
dialog_text = "Please insert a valid unique mod name. Empty mod names are not allowed. Please note the automatic stripping of invalid characters."
```

Comment (HaSa1002) [reply src/addons/libre_trainsim_modding_tools/empty_popup.tscn]: thx :)

**Review by DasCapschen — APPROVED** [review]
lgtm, thanks :)

**Comment by HaSa1002** [comment]
Cherry picked for 0.9.beta2

**Pull request merged**